drivers/sensor/tmp108/tmp108.c

```diff
@@ -0,0 +1,350 @@
+		} else if (val->val1 < 4) {
+			mode = TI_TMP108_HYSTER_2_C;
+		} else {
+			mode = TI_TMP108_HYSTER_4_C;
+		}
+
+		if (tmp108_write_config(dev,
+					TI_TMP108_HYSTER_MASK,
+					mode) < 0) {
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_ATTR_ALERT:
+		/* Spec Sheet Errata: TM is set on reset not cleared */
+		if (val->val1 == 1) {
+			mode = TI_TMP108_CONF_TM_INT;
+		} else {
+			mode = TI_TMP108_CONF_TM_CMP;
+		}
+
+		if (tmp108_write_config(dev,
+					TI_TMP108_CONF_TM_MASK,
+					mode) < 0) {
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_ATTR_LOWER_THRESH:
+		reg_value = (val->val1 << 8) | (0x00FF & val->val2);
+		if (tmp108_reg_write(dev,
+				     TI_TMP108_REG_LOW_LIMIT,
+				     reg_value) < 0) {
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_ATTR_UPPER_THRESH:
+		reg_value = (val->val1 << 8) | (0x00FF & val->val2);
+		if (tmp108_reg_write(dev,
+				     TI_TMP108_REG_HIGH_LIMIT,
+				     reg_value) < 0) {
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		if (val->val1 < 1) {
+			mode = TI_TMP108_FREQ_4_SECS;
+		} else if (val->val1 < 4) {
+			mode = TI_TMP108_FREQ_1_HZ;
+		} else if (val->val1 < 16) {
+			mode = TI_TMP108_FREQ_4_HZ;
+		} else {
+			mode = TI_TMP108_FREQ_16_HZ;
+		}
+
+		if (tmp108_write_config(dev,
+					TI_TMP108_FREQ_MASK,
+					mode) < 0) {
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_ATTR_SHUTDOWN_MODE:
+		if (tmp108_write_config(dev,
+					TI_TMP108_MODE_MASK,
+					TI_TMP108_MODE_SHUTDOWN) < 0) {
+			return -EIO;
+		}
+		drv_data->one_shot_mode = false;
+		break;
+
+	case SENSOR_ATTR_CONTINOUS_CONVERSION_MODE:
+		if (tmp108_write_config(dev,
+					TI_TMP108_MODE_MASK,
+					TI_TMP108_MODE_CONTINOUS) < 0) {
+			return -EIO;
+		}
+		drv_data->one_shot_mode = false;
+		break;
+
+	case SENSOR_ATTR_ONE_SHOT_MODE:
+		if (tmp108_write_config(dev,
+					TI_TMP108_MODE_MASK,
+					TI_TMP108_MODE_ONE_SHOT) < 0) {
+			return -EIO;
+		}
+		drv_data->one_shot_mode = true;
+		break;
+
+	case SENSOR_ATTR_ALERT_POLARITY:
+		if (val->val1 == 1) {
+			mode = TI_TMP108_CONF_POL_HIGH;
+		} else {
+			mode = TI_TMP108_CONF_POL_LOW;
+		}
+
+		if (tmp108_write_config(dev,
+					TI_TMP108_CONF_POL_MASK,
+					mode) < 0) {
+			return -EIO;
+		}
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api tmp108_driver_api = {
+	.attr_set = tmp108_attr_set,
+	.attr_get = tmp108_attr_get,
+	.sample_fetch = tmp108_sample_fetch,
+	.channel_get = tmp108_channel_get,
+	.trigger_set = tmp_108_trigger_set,
+};
+
+#ifdef  CONFIG_TMP108_ALERT_INTERRUPTS
+static int setup_interrupts(const struct device *dev)
+{
+	struct tmp108_data *drv_data = dev->data;
+
+	drv_data->gpio = device_get_binding(DT_INST_GPIO_LABEL(0, alert_gpios));
+
+	if (!drv_data->gpio) {
+		LOG_ERR("tmp108: gpio controller %s not found", DT_INST_GPIO_LABEL(0, alert_gpios));
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(drv_data->gpio, DT_INST_GPIO_PIN(0, alert_gpios), GPIO_INPUT);
```

Comment:
Similar to i2c, you can use `struct gpio_dt_spec` and associated helper functions.